### PR TITLE
Reconfigure deploy workflow to use main/latest/version number Docker tags

### PR DESCRIPTION
**Pull request by pkalita-lbl**
Fixes https://github.com/microbiomedata/issues/issues/554

These changes update the `deploy.yml` workflow file to:

* Tag Docker image with `main` for pushes to the `main` branch. Tag Docker image with `prod` and `X.Y.Z` for pushes to tags like `vX.Y.Z`. This is accomplished by using the `docker/metadata-action` action.
* Redeploy the Rancher `nmdc-dev` workloads on pushes to `main` and `nmdc` workloads on pushes to version tags. Previously only `nmdc-dev` workloads were redeployed on pushes to main. 

Once these changes are in place I'll update Rancher to point the `nmdc` workloads to the new `prod` tagged images. Then when a GitHub Release is created with a new version tag the deployment will happen automatically with no need to update anything in Rancher. This also brings this repository inline with what `nmdc-runtime` does.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -1,88 +1,87 @@
...
+        deployment: [ backend, frontend, worker ]
+
+    steps:
+      - name: Redeploy ${{ env.RANCHER_NAMESPACE }}:${{ env.IS_PROD_RELEASE && '' || 'portal-' }}${{ matrix.deployment }}
+        if: ${{ env.IS_ORIGINAL_REPO }}
+        uses: fjogeleit/http-request-action@v1
         with:
-          url: 'https://rancher2.spin.nersc.gov/v3/project/c-tmq7p:p-bkv45/workloads/deployment:nmdc-dev:portal-worker?action=redeploy'
-          method: 'POST'
+          url: https://rancher2.spin.nersc.gov/v3/project/c-tmq7p:p-bkv45/workloads/deployment:${{ env.RANCHER_NAMESPACE }}:${{ env.IS_PROD_RELEASE && '' || 'portal-' }}${{ matrix.deployment }}?action=redeploy
```
Comment (eecavanna) [inline .github/workflows/deploy.yml:84]: I see that this is where the prefix of `portal-` would be applied when needed (i.e. in the `nmdc-dev` namespace).

**Review by eecavanna — APPROVED** [review]

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (pkalita-lbl) [reply .github/workflows/deploy.yml:84]: Cool thanks. It'll be nice when they're all consistent, but no particular rush. We'll just have to remember to update this workflow again once that issue is resolved.

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -1,88 +1,87 @@
...
-        name: Login to GitHub Container Registry
-        uses: docker/login-action@v1
+          # history for all branches and tags is needed for setuptools-scm
+          fetch-depth: 0
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
```
Comment (naglepuff) [inline .github/workflows/deploy.yml:39]: What are these actions giving us? It doesn't look like the previous version of this action used these steps.

**Review by naglepuff — APPROVED** [review]
I just have one comment about the changes, which is more for educational purposes than anything else.

**Pull request merged**